File: packages/python/src/output_guard/formats/toml_.py
"""toml format: parse TOML via stdlib tomllib (Python 3.11+).

Mirror of packages/ts/src/formats/toml.ts.
"""

from __future__ import annotations

import tomllib
from typing import Any

from ..errors import ParseError


def parse_toml(text: str) -> Any:
    try:
        return tomllib.loads(text)
    except tomllib.TOMLDecodeError as e:
        raise ParseError(f"TOML parse failed: {e}", e) from e

File: packages/ts/src/formats/toml.ts
/**
 * toml format: parse TOML via `smol-toml`.
 */
import { parse as tomlParse } from "smol-toml";
import { ParseError } from "../errors.js";

export const parseToml = (text: string): unknown => {
  try {
    return tomlParse(text) as unknown;
  } catch (e) {
    throw new ParseError(`TOML parse failed: ${String(e)}`, e);
  }
};
